python/task_manager/worker_thread.py

```diff
@@ -71,44 +66,35 @@ def run(self):
             while True and self._results_dispatcher is not None:
                 # get the next task to process:
                 task_to_process = None
-                self._mutex.lock()
-                try:
+                with self._mutex:
                     while self._process_tasks and not task_to_process:
                         task_to_process = self._task
                         self._task = None
                         if not task_to_process:
                             # wait until we have something to do...
-                            self._wait_condition.wait(self._mutex)
+                            self._wait_condition.wait()
```

Comment:
Python takes the lock as a parameter in the `__init__` instead of the wait method.